File: src/main/java/szewek/fl/test/EventCounter.java
package szewek.fl.test;

import net.minecraft.item.ItemStack;
import net.minecraftforge.event.AttachCapabilitiesEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import szewek.fl.FL;

public final class EventCounter implements Runnable {
	private static final Object o = new Object();
	private static NamedCounters.Counter icount = NamedCounters.getCounter("Attaching ItemStack Capabilities");

	@SubscribeEvent
	public void count1(AttachCapabilitiesEvent<ItemStack> e) {
		synchronized (o) {
			icount.add();
		}
	}

	private static long getIcount() {
		synchronized (o) {
			return icount.getCount();
		}
	}

	@Override
	public void run() {
		boolean running = true;
		while (running) {
			try {
				Thread.sleep(5000);
			} catch (InterruptedException e) {
				e.printStackTrace();
				running = false;
			}
			FL.L.info("ItemStack count: " + getIcount());
		}
	}
}

File: src/main/java/szewek/fl/test/EventCounter.kt
package szewek.fl.test

import net.minecraft.item.ItemStack
import net.minecraftforge.event.AttachCapabilitiesEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import szewek.fl.FL

class EventCounter : Runnable {

    @SubscribeEvent
    fun count1(e: AttachCapabilitiesEvent<ItemStack>) {
        synchronized(o) {
            icount.add()
        }
    }

    override fun run() {
        var running = true
        while (running) {
            try {
                Thread.sleep(5000)
            } catch (e: InterruptedException) {
                e.printStackTrace()
                running = false
            }

            FL.L!!.info("ItemStack count: " + getIcount())
        }
    }

    companion object {
        private val o = Any()
        private val icount = NamedCounters.getCounter("Attaching ItemStack Capabilities")

        private fun getIcount(): Long {
            synchronized(o) {
                return icount.count
            }
        }
    }
}
